### PR TITLE
Refactor userPermissionBuilder

### PR DESCRIPTION
**Pull request by azlam-abdulsalam**
Improve code quality of profiles commands
- reconcile tabs while using retrieve to only fetch deployable tabs

### CODE REVIEW

File: src/impl/source/profiles/profileSync.ts [hunk 1]
```diff
@@ -108,15 +109,13 @@ export default class ProfileSync extends ProfileActions {
...
-          this.conn
-        );
+        var metadataList = await this.profileRetriever.loadProfiles(temparray);
 
         let profileWriter = new ProfileWriter();
         for (var count = 0; count < metadataList.length; count++) {
           var profileObj = metadataList[count] as Profile;
-
+          SFPowerkit.log("Reconciling  Tabs", LoggerLevel.DEBUG);
+          await this.reconcileTabs(profileObj);
```
Comment (genoud) [inline src/impl/source/profiles/profileSync.ts:118]: I assume that we are reconciling tabs here to remove some noise during retrieve. Shouldn't we do the same for other components types?
Could we reuse the reconcileTab method from the profileReconcile class?

File: src/impl/source/profiles/profileSync.ts [hunk 2]
```diff
@@ -131,12 +130,41 @@ export default class ProfileSync extends ProfileActions {
...
+      profileStatus.deleted.forEach((file) => {
         if (fs.existsSync(file)) {
           fs.unlinkSync(file);
         }
       });
     }
     return Promise.resolve(profileStatus);
   }
+
+  private async reconcileTabs(profileObj: Profile): Promise<void> {
```
Comment (genoud) [inline src/impl/source/profiles/profileSync.ts:142]: Could we use the same method from the reconcileProfile class?

**Review by genoud — COMMENTED** [review]
Nice!!

File: src/impl/source/profiles/profileSync.ts (hunk 1 above)
Comment (manisfdcsfdx) [reply src/impl/source/profiles/profileSync.ts:118]: @genoud Profile retrieve is working fine for rest of the metadata only for tab we found it is taking extra which failing on deploy again.

File: src/impl/source/profiles/profileSync.ts (hunk 2 above)
Comment (azlam-abdulsalam) [reply src/impl/source/profiles/profileSync.ts:142]: Didnt want to couple, will move to a better pattern in later refactoring

**Pull request merged**